cloudflare.php

```diff
@@ -502,3 +502,7 @@ function theme_save_pressed()
 // Enable HTTP2 Server Push
 require_once plugin_dir_path(__FILE__).'src/HTTP2ServerPush.php';
 add_action('init', array('HTTP2ServerPush', 'init'));
+
+// Display error message if user is using a subdomain
+require_once plugin_dir_path(__FILE__).'src/SubdomainChecker.php';
```

Comment:
If I understood your question correctly the answer is yes. We are using 
```
add_action('admin_notices', array('SubdomainChecker', 'enableSubdomainNotice'));
```
hook. Hooks and requests should have different structure. We added every WP hook to `cloudflare.php` which was a mess. I wanted to simplify `cloudflare.php` file and separate every functionality which uses hooks to different classes. I figured I can create `static class`es only for WP hooks. Currently I'm adding hooks to a single file under `Cloudflare-WordPress\src\`. To improve the structure we can add `Cloudflare-WordPress\src\WordPress\Hooks` which is on my todo list.